### ISSUE TITLE
deis create in a non-git repo should run with --no-remote

### ISSUE THREAD

**Issue by bacongobbler** [deis/workflow-cli#298]
As reported in https://github.com/deis/builder/issues/499, the Heroku client will run `heroku create` with `--no-remote` if the directory we are running in is not a git directory:

```
[~:dom] ⚡  heroku create
Creating app... done, ⬢ thawing-forest-56676
https://thawing-forest-56676.herokuapp.com/ | https://git.heroku.com/thawing-forest-56676.git
[~:dom] ⚡
```

Whereas with the `deis` CLI, this does not happen:

```
[~:dom] ⚡  deis create
Creating Application... done, created haptic-farmland
Error: Error when running 'git remote add deis ssh://git@deis-builder.cluster.fluid.ly:2222/haptic-farmland.git'
fatal: Not a git repository (or any of the parent directories): .git
```

We should try to mimic Heroku's workflow here, seeing how we are an "open source Heroku".

**Comment by Dakuan** [comment]
Part of why I found this confusing wasn't just the difference with Heroku's behaviour, it was also conceptual. I'm creating a deis app on a remote box - It didn't seem obvious to me that my location in a local shell would affect that. 

I though that `deis create` would have created a remote repo that I could then point to locally in a separate step if I so wished.

Example work flow
```
# create deis app for my service
$ deis create 
$ cd my-service
$ git remote add production PATH_TO_DEIS_REPO
```

**Comment by bacongobbler** [comment]
Ah I can understand that point of view. While there is a remote git repository stored at the builder, I wouldn't rely on it as a persistent git repo store for your application source code. Leave that for Github/bitbucket/your own git server etc. If the builder is rebooted, the repositories are gone because we don't have them backed by persistent disk or anything. The database WAL logs, registry images, built Heroku tarballs, and build caches are, though. The git repos in the builder are created on-the-fly when you push the local contents of the repository to the builder via `git push deis`.

Does that help clarify some things?

**Comment by Dakuan** [comment]
Absolutely, I though I never intended to use the git repo for a source control, just as a deployment pathway.

**Comment by bacongobbler** [comment]
> I thought that `deis create` would have created a remote repo that I could then point to locally in a separate step if I so wished.

For what it's worth, this is do-able, but it requires flipping on a few extra option flags (again, tooting the "Heroku compatibility" horn again):

```
$ deis create myappname --no-remote
$ deis git:remote --app myappname
```

**Comment by Cryptophobia** [comment]
This issue was moved to teamhephy/workflow-cli#9
